opentracing-api/src/main/java/io/opentracing/ActiveSpanHolder.java

```diff
@@ -0,0 +1,174 @@
+ *     subsequent calls to {@linkplain ActiveSpanHolder#active active*()}. (Helper libraries can abstract away much
+ *     of the above behind {@link java.util.concurrent.ExecutorService} wrappers)
+ *
+ * </ul>
+ */
+public abstract class ActiveSpanHolder {
+
+    /**
+     * A {@link Continuation} can be used *once* to activate a Span and some non-OpenTracing execution context (e.g.,
+     * MDC), then deactivate when processing activity moves on to another Span. (In practice, this active period
+     * typically extends for the length of a deferred async closure invocation.)
+     *
+     * <p>
+     * Most users do not directly interact with {@link Continuation}, {@link Continuation#activate()} or
+     * {@link Continuation#deactivate()}, but rather use {@link ActiveSpanHolder}-aware Runnables/Callables/Executors.
+     * Those higher-level primitives need not be defined within the OpenTracing core API, and so they are not.
+     *
+     * <p>
+     * NOTE: {@link Continuation} extends {@link java.io.Closeable} rather than AutoCloseable in order to keep support
+     * for JDK1.6.
+     *
+     * @see ActiveSpanHolder#capture(Span)
+     */
+    public static abstract class Continuation implements Closeable {
+        private final AtomicInteger refCount;
+
+        protected Continuation(AtomicInteger refCount) {
+            this.refCount = refCount;
+        }
+
+        /**
+         * Make the Span (and other execution context) encapsulated by this Continuation active and return it.
+         *
+         * <p>
+         * NOTE: It is an error to call activate() more than once on a single Continuation instance.
+         *
+         * @see ActiveSpanHolder#capture(Span)
+         * @return the newly-activated Span
+         */
+        public abstract void activate();
+
+        /**
+         * @return the {@link Span} associated with this {@link Continuation}, or null if there is no such {@link Span}.
+         */
+        public abstract Span span();
+
+        /**
+         * Mark the end of this active period for the Span previously returned by activate().
+         *
+         * <p>
+         * NOTE: It is an error to call deactivate() more than once on a single Continuation instance.
+         *
+         * @see Closeable#close()
+         */
+        public final void deactivate() {
+            doDeactivate();
+            decRef();
+        }
+
+        /**
+         * Per the {@link java.io.Closeable} API.
+         */
+        @Override
+        public final void close() {
+            this.deactivate();
+        }
+
+        /**
+         * Capture and take a reference to the {@link Span} associated with this {@link Continuation} and any 3rd-party
+         * execution context of interest.
+         *
+         * @return a new {@link Continuation} to {@link Continuation#activate()} at the appropriate time.
+         */
+        public final Continuation capture() {
```

Comment:
The idea is to make code like this easier to write:

https://github.com/bhs/opentracing-java/blob/bhs/ash_demo/opentracing-mdc-demo/src/main/java/io/opentracing/mdcdemo/TracedRunnable.java#L18

It's def just a convenience, though.

"Capture" is when the incRef happens (rather than activate), which is really important for async stuff to work correctly.